plotly/plotly/plotly.py

```diff
@@ -1520,6 +1520,77 @@ def get_dashboard_names(cls):
         return [str(dboard['filename']) for dboard in dashboards]
 
 
+class presentation_ops:
+    """
+    Interface to Plotly's Spectacle-Presentations API.
+    """
+    @classmethod
+    def upload(cls, presentation, filename, sharing='public', auto_open=True):
+        """
+        Function for uploading presentations to Plotly.
+
+        :param (dict) presentation: the JSON presentation to be uploaded. Use
+            plotly.presentation_objs.Presentation to create the presentation
+            from a string.
+        :param (str) filename: the name of the presentation to be saved in
+            your Plotly account. Will overwrite a presentation of the same
+            name if it already exists in your files.
+        :param (str) sharing: can be set to either 'public', 'private'
+            or 'secret'. If 'public', your presentation will be viewable by
+            all other users. If 'private' only you can see your presentation.
+            If 'secret', the url will be returned with a sharekey appended
```

Comment:
I'm not sure if folks outside of Plotly will know what "sharekey" is. Could this be: "The url will contain a random, hard-to-guess string appended to it. Anyone with the url may view the presentation."

Reply:
How about i keep the word "sharekey" but explain what it is? I've used the term in `dashboard_ops` function as well and I think it's a good term for people to know

Reply:
Sure. What do you propose?

Reply:
How about: If it is set to 'secret', the url will be returned with a string of random characters appended to the url which is called a sharekey. The point of a sharekey is that it makes the url very hard to guess, but anyone with the url can view the presentation.